File: java/src/main/java/kr/latera/baseball/domain/RandomNumberGenerator.java
package kr.latera.baseball.domain;

import java.util.Random;

public class RandomNumberGenerator implements NumberGenerator {

    private final Random random;
    private final int min;
    private final int max;

    public RandomNumberGenerator(int minInclusive, int maxExclusive) {
        random = new Random();
        this.min = minInclusive;
        this.max = maxExclusive;
    }

    @Override
    public int generate() {
        return random.nextInt(max - min) + min;
    }
}

File: baseball/RandomNumberGenerator.go
package baseball

import (
	"math/rand"
)

// RandomNumberGenerator implements NumberGenerator using ranom module
type randomNumberGenerator struct {
	min int
	max int
}

// Generate returns random number [min, max]
func (G *randomNumberGenerator) Generate() int {
	return rand.Intn(G.max-G.min) + G.min
}

// NewRandomNumberGenerator is factory method for number random number generator
func NewRandomNumberGenerator() NumberGenerator {
	generator := new(randomNumberGenerator)
	generator.min = NumberMin
	generator.max = NumberMax
	return generator
}
